crates/ark-metadata/src/metadata_manager.rs

```diff
@@ -171,16 +171,27 @@ impl<'a, T: Storage, C: StarknetClient, F: FileManager> MetadataManager<'a, T, C
                     is_cache_updated: false,
                 })
             }
-            _ => {
-                debug!("Fetching image... {}", url);
-                let response = reqwest::get(url).await?;
+            ImageCacheOption::Save => {
+                let response = self
+                    .request_client
+                    .get(url)
+                    .timeout(Duration::from_secs(5))
+                    .send()
+                    .await?;
+
                 let headers = response.headers().clone();
                 let bytes = response.bytes().await?;
                 let (content_type, content_length) = extract_metadata_from_headers(&headers)?;
+                let file_ext = file_extension_from_mime_type(content_type.as_str());
+
+                debug!(
+                    "Image: Content-Type={}, Content-Length={}, File-Ext={}",
+                    content_type, content_length, file_ext
+                );
 
                 self.file_manager
                     .save(&FileInfo {
-                        name: format!("{}.{}", token_id.to_hex(), file_ext),
+                        name: format!("{}.{}", token_id.to_decimal(false), file_ext),
```

Comment:
Saving in hex would be better for formatting and consistency. Anyway in lambdas APIs we have functions to convert a decimal or unpadded hex into correctly padded hex.

Reply:
wdyt @kwiss @YohanTz